### PR TITLE
Address CVE-2023-29383 in shadow-utils

### PR DESCRIPTION
**Pull request by kanikanema**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Fixes CVE-2023-29383

###### Change Log  
This fix is based on https://github.com/NixOS/nixpkgs/pull/233924. An earlier attempt that included just the first patch caused a build break.

###### Does this affect the toolchain?  
No

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-29383

###### Test Methodology
- Buddy Build: 424052. (Please note the tdnf install step fails due to a bug in InstallPackagesTdnf.sh)
- Full Pipeline build id: 424088

### CODE REVIEW

File: SPECS/shadow-utils/shadow-utils.spec [hunk 1]
```diff
@@ -176,6 +177,9 @@ chmod 000 %{_sysconfdir}/shadow
 %{_libdir}/libsubid.so
 
 %changelog
+* Wed Sept 20 2023 Kanika Nema <kanikanema@microsoft.com> - 4.9-13
```
Comment (mfrw) [inline SPECS/shadow-utils/shadow-utils.spec:180]: Typo
```suggestion
* Wed Sep 20 2023 Kanika Nema <kanikanema@microsoft.com> - 4.9-13
```

File: SPECS/shadow-utils/shadow-utils.spec [hunk 2]
```diff
@@ -22,6 +22,7 @@ Source12:       useradd-default
 Source13:       login-defs
 Patch0:         chkname-allowcase.patch
 Patch1:         libsubid-pam-link.patch
+Patch2:         CVE-2023-29383.patch
```
Comment (mfrw) [inline SPECS/shadow-utils/shadow-utils.spec:25]: I do not think this spec uses `%autosetup`, we might need to add an explicit `%patch2 -p1`

Comment (kanikanema) [reply SPECS/shadow-utils/shadow-utils.spec:25]: Ah yes! Good catch. Will fix and rerun the builds.

**Review by jslobodzian — APPROVED** [review]

File: SPECS/shadow-utils/shadow-utils.spec [hunk 3]
```diff
@@ -176,6 +178,9 @@ chmod 000 %{_sysconfdir}/shadow
 %{_libdir}/libsubid.so
 
 %changelog
+* Wed Sep 20 2023 Kanika Nema <kanikanema@microsoft.com> - 4.9-13
```
Comment (mfrw) [inline SPECS/shadow-utils/shadow-utils.spec:181]: NIT: 
I would request to take the suggestions from the linter:
At-least the ones that:
`/usr/sbin` -> %{_sbindir}`

**Review by mfrw — APPROVED** [review]
LGTM

File: SPECS/shadow-utils/shadow-utils.spec (hunk 3 above)
Comment (kanikanema) [reply SPECS/shadow-utils/shadow-utils.spec:181]: Sorry, didn't pick up this comment. The builds have completed, and I didn't want to update the spec file and not run a build (even if it is just tp take in lint changes). Thanks for the comment, I will look through the lint errors a bit more carefully next time.

**Pull request merged**